ui/src/app/applications/components/filter/filter.tsx

```diff
@@ -17,6 +17,40 @@ interface FilterProps {
     radio?: boolean;
 }
 
+export interface CheckboxOption {
+    label: string;
+    count?: number;
+    icon?: React.ReactNode;
+}
+
+export const CheckboxRow = (props: {value: boolean; onChange?: (value: boolean) => void; option: CheckboxOption}) => {
+    const [value, setValue] = React.useState(props.value);
+
+    React.useEffect(() => {
+        setValue(props.value);
+    }, [props.value]);
+
+    return (
+        <div className={`checkbox__item ${value ? 'checkbox__item--selected' : ''}`} onClick={() => setValue(!value)}>
```

Comment:
I think `checkbox__item` and `checkbox__item--selected` logically should be part of this filters component: `filter__item`, `filter__item--selected`. Can you please move these styles from https://github.com/argoproj/argo-ui/blob/fecc293fb19f4c00891d8470932fb4951e21834c/v2/components/checkbox/checkbox.scss#L15 as well?